packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverButton/TeachingPopoverButton.types.ts

```diff
@@ -1,17 +1,30 @@
-import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import type { ButtonProps, ButtonSlots, ButtonState } from '@fluentui/react-button';
 
-export type TeachingPopoverButtonSlots = {
-  root: Slot<'div'>;
-};
-
-/**
- * TeachingPopoverButton Props
+/*
+ * These buttons are the same as a default fluent v9 button
+ * with overridden style/functionality based on Teaching Context
  */
-export type TeachingPopoverButtonProps = ComponentProps<TeachingPopoverButtonSlots> & {};
+export type TeachingPopoverButtonSlots = ButtonSlots;
 
-/**
- * State used in rendering TeachingPopoverButton
- */
-export type TeachingPopoverButtonState = ComponentState<TeachingPopoverButtonSlots>;
-// TODO: Remove semicolon from previous line, uncomment next line, and provide union of props to pick from TeachingPopoverButtonProps.
-// & Required<Pick<TeachingPopoverButtonProps, 'propName'>>
+export type TeachingPopoverButtonType = 'primary' | 'secondary';
+
+export type TeachingPopoverButtonProps = ButtonProps & {
+  /*
+   * Button type, compulsory to be provided by user.
+   */
+  buttonType: TeachingPopoverButtonType;
```

Comment:
I am curious how is that different to `appearance` prop that already contains `secondary` & `primary`.

Reply:
We are intending to use primary/secondary button types here, however they have different contextual meanings based on whether the carousel is present or not, this enables the user to provide the buttons definition as primary/secondary but have them contextually styled and functional depending on the composition.

Because of this, we don't necessarily want the underlying button appearance type to be 1:1 as there may be situations where it is flipped internally.

Reply:
Resolution tracked by #30127.